api/graphql/reservation_units/reservation_unit_mutations.py

```diff
@@ -213,6 +220,47 @@ class ReservationUnitUpdateMutation(
         else (AllowAny,)
     )
 
+    def remove_personal_data_and_logs_on_archive(input):
+        """When reservation unit is archived, we want to delete all personally identifiable information (GDPR stuff).
+        Because all changes are stored to the audit log, we also need to delete old audit events related to the unit.
+        """
+        if "is_archived" not in input:
+            return
+
+        reservation_unit_pk = input["pk"]
+        reservation_unit = ReservationUnit.objects.get(pk=reservation_unit_pk)
+        old_archived_value = reservation_unit.is_archived
+        new_archived_value = input["is_archived"]
+
+        if not new_archived_value or old_archived_value == new_archived_value:
+            return
+
+        # Don't allow new contact information to be saved when unit is archived
+        if "contact_information" in input:
+            del input["contact_information"]
+
+        # Reset contanct information
+        reservation_unit.contact_information = ""
+        reservation_unit.save()
+
+        # Delete audit log entries
+        content_type_id = ContentType.objects.get(
+            app_label="reservation_units", model="reservationunit"
+        ).id
+        deleted_log_entires = LogEntry.objects.filter(
```

Comment:
`LogEntry.objects.get_for_object(reservation_unit)` also works. And do not need the upper `content_type_id`

Reply:
Noice! I read the documentation but somehow managed to miss that one. Shame that VSCode intellisense is so bad with Python 😞